kubetest2/internal/deployers/eksapi/deployer.go

```diff
@@ -69,6 +71,7 @@ type deployerOptions struct {
 	KubernetesVersion           string        `flag:"kubernetes-version" desc:"cluster Kubernetes version"`
 	NodeReadyTimeout            time.Duration `flag:"node-ready-timeout" desc:"Time to wait for all nodes to become ready"`
 	Nodes                       int           `flag:"nodes" desc:"number of nodes to launch in cluster"`
+	OsDistro                    string        `flag:"os-distro" desc:"Specifies the OS distribution for the AMI. Allowed values: ['al2', 'al2023'] (case-insensitive)"`
```

Comment:
Can we make this option a little more targeted to this specific behavior, something like `--node-name-strategy`?

We'll eventually have to add `--ami-type` to test with MNG, but that concept doesn't mean anything for an unmanaged nodegroup.

Reply:
Do you mean we need to decouple the behaviors from the os-distro? Option like naming strategy would allow use cases like AL2 + SessionName and AL2023 + PrivateDNSName. Although we can control it via pipeline, is there any other user will use it in that way?

Reply:
> Option like naming strategy would allow use cases like AL2 + SessionName and AL2023 + PrivateDNSName.

Yeah exactly. When we add support for AL2023 managed node groups, this tool will need an option for `--ami-type`, and we will set `--node-name-strategy` based on *that* value; but for unmanaged nodes, the AMI may use `bootstrap.sh`, `nodeadm`, or something else entirely. I want to avoid making too many assumptions about feature sets here, and just expose the individual options. Our higher level test framework that uses this tool can centralize all the details of which AMI's support which feature, how to test a given AMI, etc.